dbos/_sys_db.py

```diff
@@ -1650,7 +1650,7 @@ def start_queued_workflows(
                     return []
 
             # Compute max_tasks, the number of workflows that can be dequeued given local and global concurrency limits,
-            max_tasks = float("inf")
+            max_tasks = 100  # To minimize contention with large queues, never dequeue more than 100 tasks
```

Comment:
I'm worried about this hard cap -- the contention is not so much about large queues but about large amounts of concurrent workers. This cap will eventually come as a surprise to users.

Reply:
The cap significantly improves stability under high load, and has no effect under low load. We'll likely adjust it over time, but it would not be helpful to remove it.

Reply:
What I'm worried about is the hardcoded part :)